### PR TITLE
Add "join table" for visits

### PR DESCRIPTION
**Pull request by DevaWinner**
In this pull request, I introduce significant changes to the database structure, including the creation of the `vets`, `specializations`, and `visits` tables. These changes facilitate the representation of relationships between vets, species, and animals, as well as tracking visits and specialties. SQL queries have been written to address specific questions related to these tables. Screenshots of the query results have been included in the Pull Request description for reference.

### Changes Made:

- Created a table named `vets` with columns: `id` (autoincremented PRIMARY KEY), `name` (string), `age` (integer), and `date_of_graduation` (date).
- Created a "join table" named `specializations` to handle the many-to-many relationship between vets and species.
- Created a "join table" named `visits` to handle the many-to-many relationship between animals and vets, including the date of the visit.
- Inserted data into the `vets` table for four veterinarians with their respective information.
- Inserted data into the `specializations` table to represent the specialties of certain vets.
- Inserted data into the `visits` table to represent visits of various animals to different vets.

### Screenshots:
Please reference comments for updated screenshots

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -139,3 +139,85 @@ JOIN animals a ON o.id = a.owner_id
...
+    SELECT s.name AS specialty_name, COUNT(*) AS visit_count
+    FROM visits v
+    JOIN animals a ON v.animal_id = a.id
+    JOIN vets ve ON v.vet_id = ve.id
+    JOIN specializations sp ON ve.id = sp.vet_id
+    JOIN species s ON sp.species_id = s.id
+    WHERE a.name = 'Maisy Smith'
+    GROUP BY s.name
+    ORDER BY visit_count DESC
+    LIMIT 1;
```
Comment (abdulhamiid) [inline queries.sql]: Please add the correct `query` and the `screenshot` 🤗🛠

image

File: queries.sql [hunk 2]
```diff
@@ -139,3 +139,85 @@ JOIN animals a ON o.id = a.owner_id
...
+    JOIN animals a ON v.animal_id = a.id
+    JOIN vets vt ON v.vet_id = vt.id
+    WHERE vt.name = 'Stephanie Mendez';
+
+    --  List all vets and their specialties, including vets with no specialties.
+    SELECT vt.name, COALESCE(array_agg(s.name), ARRAY[]::text[]) AS specialties
+    FROM vets vt
+    LEFT JOIN vet_specialties vs ON vt.id = vs.vet_id
+    LEFT JOIN specialties s ON vs.specialty_id = s.id
+    GROUP BY vt.id, vt.name;
```
Comment (abdulhamiid) [inline queries.sql]: Please add the screenshot of the query result 👀

image

File: queries.sql [hunk 3]
```diff
@@ -139,3 +139,85 @@ JOIN animals a ON o.id = a.owner_id
...
+    ORDER BY visit_count DESC
+    LIMIT 1;
+
+    --  Who was Maisy Smith's first visit?
+    SELECT a.name AS animal_name, MIN(v.visit_date) AS first_visit_date
+    FROM visits v
+    JOIN animals a ON v.animal_id = a.id
+    JOIN vets vt ON v.vet_id = vt.id
+    WHERE vt.name = 'Maisy Smith'
+    GROUP BY a.name;
```
Comment (abdulhamiid) [inline queries.sql]: Please add the correct `query` and the `screenshot` 🤗🛠

image

**Review by abdulhamiid — CHANGES_REQUESTED** [review]
# Status: Required Changes :recycle:
Hi @DevaWinner 👋, 
Alt Text
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights 🌟

- ✔️ Descriptions are descriptive
- ✔️ Understand the different types of relationships between tables.
- ✔️ Use primary key & foreign key mechanisms for joining tables.

## Required Changes ♻️
_Please kindly check the comments under the review._

## Optional suggestions 🤷‍♂️
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 1 above)
Comment (DevaWinner) [reply queries.sql]: i must have mixed up the screenshots, let me get it sorted.

Thank you

File: queries.sql (hunk 2 above)
Comment (DevaWinner) [reply queries.sql]: Thank you for noticing the mix with the screenshots. I will sort them now

**Comment by DevaWinner** [comment]
## Please reference below for updated screenshots

### 1. Who was the last animal seen by William Tatcher?
Screenshot from 2023-08-25 08-26-51

### 2. How many different animals did Stephanie Mendez see?
Screenshot from 2023-08-25 08-27-11

### 3. List all vets and their specialties, including vets with no specialties.
Screenshot from 2023-08-25 08-27-25

### 4. List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
Screenshot from 2023-08-25 08-27-35

### 5. What animal has the most visits to vets?
Screenshot from 2023-08-25 08-27-57

### 6. Who was Maisy Smith's first visit?
Screenshot from 2023-08-25 08-28-28

### 7. Details for most recent visit: animal information, vet information, and date of visit.
Screenshot from 2023-08-25 08-28-40

### 8. How many visits were with a vet that did not specialize in that animal's species?
Screenshot from 2023-08-25 08-28-53

### 9. What specialty should Maisy Smith consider getting? Look for the species she gets the most
Screenshot from 2023-08-25 08-29-17

**Review by Strangeal — APPROVED** [review]
Hi @DevaWinner, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 






## Highlights
- Added queries screenshot  ✅ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**